pkgs/applications/editors/vscode/generic.nix

```
@@ -109,7 +109,7 @@ let
       gappsWrapperArgs+=(
         # Add gio to PATH so that moving files to the trash works when not using a desktop environment
         --prefix PATH : ${glib.bin}/bin
-        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--enable-features=UseOzonePlatform --ozone-platform=wayland}}"
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}}"
```

```suggestion
        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```
This does not work? Maybe with auto detection we could remove the condition on NIXOS_OZONE_WL?

---

Yes, `auto` also works and enables wayland correctly. I'm not sure it should be enabled by default though, I think it's still considered experimental. In vscode at least.

---

I _believe_ I read somewhere that `auto` is eventually going to be enabled on some Electron release so I'd prefer that if verified but I don't see anything wrong here.